app/src/main/java/fi/aalto/legroup/achso/entities/OptimizedVideo.java

```diff
@@ -0,0 +1,359 @@
+ * You should use this class directly as much as possible and inflating to Video only when necessary.
+ *
+ * These should round trip:
+ *   new OptimizedVideo(video).inflate()
+ *   new OptimizedVideo(optimizedVideo.inflate())
+ */
+public class OptimizedVideo {
+
+    private transient String manifestUri;
+    private transient VideoRepository repository;
+
+    private String videoUri;
+    private String thumbUri;
+    private UUID id;
+    private String title;
+    private String genre;
+    private String tag;
+    private long dateInMs;
+    private long lastModifiedInMs;
+    private double locationLatitude;
+    private double locationLongitude;
+    private float locationAccuracy;
+    private boolean hasLocation;
+    private boolean hasLastModified;
+
+    // This stores all the annotation text data. The single annotations refer
+    // to parts of this buffer with indices specified in annotationTextStartEnd.
+    private String annotationTextBuffer;
+
+    // Index to the User pool
+    private int authorUserIndex;
+
+    // Annotations
+    // X and Y coordinates are stored in interleaved pairs
+    // Text is stored in (begin, end) pairs to annotationTextBuffer
+    // Author indices are to user pool
+    private long[] annotationTime;
+    private float[] annotationXY;
+    private int[] annotationTextStartEnd;
+    private int[] annotationAuthorUserIndex;
+
+    public UUID getId() {
+        return id;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public String getGenre() {
+        return genre;
+    }
+
+    public Uri getVideoUri() {
+        return Uri.parse(videoUri);
+    }
+
+    public Uri getThumbUri() {
+        return Uri.parse(thumbUri);
+    }
+
+    public Uri getManifestUri() {
+        return Uri.parse(manifestUri);
+    }
+
+    public Location getLocation() {
+        if (!hasLocation) {
+            return null;
+        }
+
+        Location location = new Location("optimized getter");
+        location.setLatitude(locationLatitude);
+        location.setLongitude(locationLongitude);
+        location.setAccuracy(locationAccuracy);
+        return location;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    public void setTag(String tag) {
+        this.tag = tag;
+    }
+
+    public int getAnnotationCount() {
+        return annotationTime.length;
+    }
+
+    public String getAnnotationText(int i) {
+
+        // Annotation text data is stored in one giant buffer and is referenced with indices
+        // The indices are stored in { Start0, End0, Start1, End1, ...} interleaved array
+        int start = annotationTextStartEnd[i * 2];
+        int end = annotationTextStartEnd[i * 2 + 1];
+
+        if (start < 0 || end < 0) {
+
+            // Mark null with negative indices
+            return null;
+        } else if (start == end) {
```

Comment:
Would this be slightly more readable without the `else` blocks since it uses early return?